pychromecast/controllers/plex.py

```diff
@@ -1,40 +1,538 @@
+                msg["current_time"] = offset_now
+
+            self._send_cmd(
+                msg,
+                namespace="urn:x-cast:com.google.cast.media",
+                inc_session_id=True,
+                inc=False,
+            )
+        else:
+            self.logger.debug(
+                "Cant reset teh stream as _last_play_msg "
+                "is not set by _send_start_play")
+
+    def _send_start_play(self, media=None, **kwargs):
+        """Helper to send a playback command.
+
+        Args:
+            media (None, optional): :class:`~plexapi.base.Playable
+            **kwargs: media_to_chromecast_command docs string.
+        """
+        msg = media_to_chromecast_command(
+            media, requestiId=self._inc_request(), **kwargs
+        )
+        self.logger.debug("Create command: \n%r\n",
+                          json.dumps(msg, indent=4))
+        self._last_play_msg = msg
+        self._send_cmd(
+            msg,
+            namespace="urn:x-cast:com.google.cast.media",
+            inc_session_id=True,
+            inc=False,
+        )
+
+    def block_until_playing(self, media=None, timeout=None, **kwargs):
+        """Block until this playing, typically usefull in a script
+
+           another way to the the same is the check if the
+           controllers is_active or use self.status.player_state
+
+           Args:
+            media (None, optional): Can also be :class:`~plexapi.base.Playable
+                                   if its not, you need to fill out all the kwargs.
+            timeout (None, int): default None
+            **kwargs: See media_to_chromecast_command docs string.
+
+        """  # noqa
+        # Incase media isnt playing
+        self.play_media_event.clear()
+        self.play_media(media, **kwargs)
+        self.play_media_event.wait(timeout)
+        self.play_media_event.clear()
+
+    def play_media(self, media=None, **kwargs):
+        """Start playback on the chromecast
+
+        Args:
+            media (None, optional): Can also be :class:`~plexapi.base.Playable
+                                   if its not, you need to fill out all the kwargs.
+            **kwargs: See media_to_chromecast_command docs string.
+        """  # noqa
+        self.play_media_event.clear()
+
+        def app_launched_callback():  # pylint: disable=missing-docstring
+            try:
+                self._send_start_play(media, **kwargs)
+            finally:
+                self.play_media_event.set()
+
+        self.launch(app_launched_callback)
+
+    def join(self, timeout=None):
+        """Join the thread."""
+        self._socket_client.join(timeout=timeout)
+
+    def disconnect(self, timeout=None, blocking=True):
+        """Disconnect the controller."""
+        self._socket_client.disconnect()
+        if blocking:
+            self.join(timeout=timeout)
+
+
+# pylint: disable=too-many-public-methods
+class PlexApiController(PlexController):
```

Comment:
Why does this class exist? `self.pms` is not being used.

Reply:
Its being used inside _get_current_media, I have separated this a in its own class as I haven't found another way to to do the public methods in this class without plexapi.